test/runtests.jl

```diff
@@ -122,26 +126,26 @@ end
 
 # Empty arrays
 test_imap(
-  Any[],
```

Comment:
This is really not designed to test that. What adding those would really be testing is `Union{}[] == Any[]` which is not in the scope of Iterators.